packages/client/hmi-client/src/components/workflow/ops/model-config/tera-model-config-drilldown.vue

```diff
@@ -639,15 +634,28 @@ const onSelectConfiguration = async (config: ModelConfiguration) => {
 	if (pdfViewer.value && config.extractionPage) {
 		pdfViewer.value.goToPage(config.extractionPage);
 	}
-	// Checks if there are unsaved changes to current model configuration
-	if (isModelConfigsEqual(originalConfig.value, knobs.value.transientModelConfig)) {
+	// If no changes were made switch right away
+	if (isModelConfigsEqual(originalConfig, knobs.value.transientModelConfig)) {
 		applyConfigValues(config);
 		return;
 	}
 
+	const lostItems: string[] = [];
+
+	if (!isModelConfigValuesEqual(knobs.value.transientModelConfig, originalConfig)) lostItems.push('values');
+	if (knobs.value.transientModelConfig.name !== originalConfig?.name) lostItems.push('name');
+	if (knobs.value.transientModelConfig.description !== originalConfig?.description) lostItems.push('description');
+
+	let lostItemsStr = '';
+	if (lostItems.length > 1) {
+		lostItemsStr = `${lostItems.slice(0, -1).join(', ')} and ${lostItems.slice(-1)}`;
+	} else {
+		lostItemsStr = lostItems[0];
+	}
```

Comment:
- There are a handful of `knobs.value.transientModelConfig` repeated here, can you you stick that into a variable so the code is easier to read?
- The 'and-string concatenation`, let's put that into util/text.ts, also you seem to be missing the case to handle an empty array.